### PR TITLE
New submodules: ampcombi parsetables/complete/cluster

### PR DESCRIPTION
**Pull request by Darcy220606**
## PR checklist

For this PR to PASS it needs approval of the new minimal test datasets #1189

This PR addresses the issue #4999 and closes PRs #4659 and #4533

- [x] This comment contains a description of changes (with reason).
- [x] If you've fixed a bug or added code that should be tested, add tests!
- [x] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [x] If necessary, include test data in your PR.
- [x] Remove all TODO statements.
- [x] Emit the `versions.yml` file.
- [x] Follow the naming conventions.
- [x] Follow the parameters requirements.
- [x] Follow the input/output options guidelines.
- [x] Add a resource `label`
- [x] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - For modules:
    - [x] `nf-core modules test <MODULE> --profile docker`
    - [x] `nf-core modules test <MODULE> --profile singularity`
    - [x] `nf-core modules test <MODULE> --profile conda`
  - For subworkflows:
    - [ ] `nf-core subworkflows test <SUBWORKFLOW> --profile docker`
    - [ ] `nf-core subworkflows test <SUBWORKFLOW> --profile singularity`
    - [ ] `nf-core subworkflows test <SUBWORKFLOW> --profile conda`

### CODE REVIEW

**Comment by Darcy220606** [comment]
Note: For all the nf-tests, I asserted for the main process either the names of the folders/files because they all depend on a downloaded database that change the md5 sums.

**Comment by Darcy220606** [comment]
Hey @louperelo , do you know which pandas version was used i the old version of ampcombi ? The thing is its running locally but in github actions it seems like its running a new version.

**Review by pinin4fjords — CHANGES_REQUESTED** [review]
First pass review here. A few common themes:

- Checking file names is a bit of a risky way to validate a process has run. It can be OK for a small subset of outputs, but if you do it with everything then you're not checking very hard that things have run well, you're not doing much more than what a stub run would do. Snapshot file content wherever possible (sometimes there's a random seed you can set for reproducibility), but failing that try to identify some diagnostic strings from the content and check as suggested in the nf-test docs I linked to.
- You can't just delete a module- people could be using it! You can deprecate though- I've provided a link.
- Make sure you indent for reproducibility.
- Order the meta.ymls to match the main.nfs.
- Use meta maps on the file inputs and outputs wherever possible.

Happy to come back and take another look after revisions.

File: modules/nf-core/ampcombi/complete/tests/main.nf.test
Comment (jasmezz) [inline modules/nf-core/ampcombi/complete/tests/main.nf.test:74]: Check for file contents if possible?

File: modules/nf-core/ampcombi/parsetables/meta.yml
Comment (jasmezz) [inline modules/nf-core/ampcombi/parsetables/meta.yml:21]:
```suggestion
      description: "A parsing tool to convert and summarise the outputs from multiple AMP detection tools in a standardized format."
```

File: modules/nf-core/ampcombi/parsetables/meta.yml
Comment (jasmezz) [inline modules/nf-core/ampcombi/parsetables/meta.yml:35]:
```suggestion
      description: The path to the directory containing the results for the AMP tools for each processed sample or a list of files corresponding to each file generated by AMP tools.
```

File: modules/nf-core/ampcombi/parsetables/meta.yml
Comment (jasmezz) [inline modules/nf-core/ampcombi/parsetables/meta.yml:43]:
```suggestion
      description: The path to the file corresponding to the respective annotated files with either '.gbk' or '.gbff' extensions. File names must contain the corresponding sample name, i.e. sample_1.faa where "sample_1" is the sample name.
```

File: modules/nf-core/ampcombi/parsetables/tests/main.nf.test
Comment (jasmezz) [inline modules/nf-core/ampcombi/parsetables/tests/main.nf.test:21]:
```suggestion
                amp_input = [
                    [id:'sample_1'],
                    [
                        file('https://github.com/nf-core/test-datasets/raw/modules/data/delete_me/ampcombi/test_files_0.2/ampir/sample_1/sample_1.ampir.tsv', checkIfExists: true),
                        file('https://github.com/nf-core/test-datasets/raw/modules/data/delete_me/ampcombi/test_files_0.2/amplify/sample_1/sample_1.amplify.tsv', checkIfExists: true)
                    ]
                ]
```

File: modules/nf-core/ampcombi/parsetables/tests/main.nf.test
Comment (jasmezz) [inline modules/nf-core/ampcombi/parsetables/tests/main.nf.test:56]:
```suggestion
                amp_input = [
                    [id:'sample_1'],
                    [
                        file('https://github.com/nf-core/test-datasets/raw/modules/data/delete_me/ampcombi/test_files_0.2/ampir/sample_1/sample_1.ampir.tsv', checkIfExists: true),
                        file('https://github.com/nf-core/test-datasets/raw/modules/data/delete_me/ampcombi/test_files_0.2/amplify/sample_1/sample_1.amplify.tsv', checkIfExists: true)
                    ]
                ]
```

**Review by jasmezz — CHANGES_REQUESTED** [review]
What @pinin4fjords says and (as you know from me), many minor typo/alignment style suggestions. :grin:

File: modules/nf-core/ampcombi/main.nf
Comment (Darcy220606) [reply modules/nf-core/ampcombi/main.nf:1]: When i add the deprecation the nf-tests dont pass through, and so what i did was i added the nf-tests for ampcombi (the old version), ran the nf-tests successfully and then added the deprecation message. But im afraid this old version wont pass the github action tests. 

Also to leave the old version of ampcombi, i had to rename the new one under another directory name `ampcombifilter` to differentiate between them to run the nf-tests. Is that ok you think ?

File: modules/nf-core/ampcombi/cluster/main.nf
Comment (Darcy220606) [reply modules/nf-core/ampcombi/cluster/main.nf]: We dont need to identify the sample ID in both `ampcombi/complete` and `ampcombi/cluster`, because in teh original module it was devloped that it takes a list of files (the result from `ampcombi/parsetables`) and so when implemented in a pipeline all the files will be mixed in a channel and then input into the submodules as a list that doesnt take into account the sample ID.

File: modules/nf-core/ampcombi/cluster/meta.yml
Comment (Darcy220606) [reply modules/nf-core/ampcombi/cluster/meta.yml]: done

File: modules/nf-core/ampcombi/cluster/tests/main.nf.test
Comment (Darcy220606) [reply modules/nf-core/ampcombi/cluster/tests/main.nf.test]: Checked the content now in all submodules

File: modules/nf-core/ampcombi/cluster/tests/nextflow.config
Comment (Darcy220606) [reply modules/nf-core/ampcombi/cluster/tests/nextflow.config]: this is now not required as i dont need the setup processes anymore.

File: modules/nf-core/ampcombi/complete/main.nf
Comment (Darcy220606) [reply modules/nf-core/ampcombi/complete/main.nf]: These two submodules `ampcombi/complete`and `ampcombi/cluster` are not meant to be used more than once in a workflow. They are summarizing tools that take the channels from all 'mixed' channels and merges them together. Therefore, no meta is required.

File: modules/nf-core/ampcombi/cluster/tests/main.nf.test
Comment (Darcy220606) [reply modules/nf-core/ampcombi/cluster/tests/main.nf.test]: Ok so i have now added minimal test datasets for `ampcombi/complete` in a PR https://github.com/nf-core/test-datasets/pull/1189. And then i use `ampcombi/complete` output as a setup for `ampcombi/cluster`.

File: modules/nf-core/ampcombi/complete/tests/nextflow.config
Comment (Darcy220606) [reply modules/nf-core/ampcombi/complete/tests/nextflow.config]: This is now removed as the `ampcombi/parsetables` are not in the setup anymore.

File: modules/nf-core/ampcombi/ampcombi/tests/main.nf.test
Comment (Darcy220606) [reply modules/nf-core/ampcombi/ampcombi/tests/main.nf.test]: Now added chacks for complete line numbers or strings.

File: modules/nf-core/ampcombi2/parsetables/tests/nextflow.config
Comment (jasmezz) [inline modules/nf-core/ampcombi2/parsetables/tests/nextflow.config:6]:
```suggestion
```

**Review by jasmezz — APPROVED** [review]
LGTM, only tiny formatting stuffs. :smiley_cat:

File: modules/nf-core/ampcombi2/complete/tests/main.nf.test
Comment (Darcy220606) [reply modules/nf-core/ampcombi2/complete/tests/main.nf.test:12]: well spotted

**Comment by SPPearce** [comment]
Is there anything outstanding to merge this in @Darcy220606 ?

**Comment by Darcy220606** [comment]
> Is there anything outstanding to merge this in @Darcy220606 ?

@SPPearce Ive addressed all the suggestions already, so it should be ready to merge once dev is merged in.

**Pull request merged**
